application/models/Permission.php

```diff
@@ -566,16 +566,15 @@ public function hasPermission($iEntityID, $sEntityName, $sPermission, $sCRUD = '
         }
         $sCRUD = $sCRUD . '_p';
 
-        /* Always return false for guests */
-        // TODO: should not be necessary
+        /* Be sure to have an user id */
         $iUserID = $this->getUserId($iUserID);
-        if (!$iUserID && $iUserID !== 0) {
+        /* Always return false for guests */
+        if (!$iUserID) {
```

Comment:
User id === 0 or === null?

Reply:
empty … === 0 or is null or === "" empmty mean empty :)

See https://github.com/LimeSurvey/LimeSurvey/pull/1755/files#diff-10077a3f8791500882779e35a827caf5e6578d5140625094d224de5e3c166780R732

In 3.X : we **never** send 0 to the function …

Reply:
Writing "empty($userId)" is more clear than "!$userId", I think. Or?

Reply:
K